membership/members.lua

```diff
@@ -87,8 +88,9 @@ function members.set(uri, status, incarnation, payload)
     _all_members[uri] = {
         status = status,
         incarnation = incarnation,
-        payload = payload or (member or {}).payload,
+        payload = (params or {}).payload or (member or {}).payload,
```

Comment:
Let's make this part more economical for the sake of gc: instead of creating empty tables write a couple of ifs afterwards